src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -434,5 +435,18 @@ internal static void SetCurrentUICulture(CultureInfo cultureInfo)
                 Thread.CurrentUICulture.SetValue(thread, cultureInfo);
             }
         }
+
+        internal static class Assembly
+        {
+            private const string TypeName = "System.Reflection.Assembly";
+
+            internal static readonly Type Type = ReflectionUtil.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            internal static readonly MethodInfo GetTypeThrowOnErrorIgnoreCase = Type
```

Comment:
I agree it's not great as a call site annotation.  But this is only temporary.  Once we switch to 4.6 these will become named arguments.